lib/buffer.js

```diff
@@ -379,6 +378,16 @@ Buffer.prototype.toString = function(encoding, start, end) {
 };
 
 
+Buffer.prototype.toString = function() {
+  const length = this.length | 0;
+  if (length === 0)
+    return '';
+  if (arguments.length === 0)
+    return this.utf8Slice(0, length);
+  return slowToString(this, arguments[0], arguments[1], arguments[2]);
```

Comment:
This seems to be a tiny bit faster than `slowToString.apply(this, arguments)` but that may change when we upgrade to V8 4.4.